### DISCUSSION TITLE
Possibility to read value from registry

### DISCUSSION THREAD

**Discussion by JajajTec** [Rem0o/FanControl.Releases#1095] [Q&A]
hey there!
i wanted to ask if there is any way i can read a value from registry and use it as a costume sensor?
that would make my life a whole lot easyer and enable compatibility with for excample aida64 or a external sensor.
that way you could theoretically include battery voltage, cpu/gpu package power and many other things!

**Comment by Rem0o** [comment]
You can do it via a Plugin. That's how https://github.com/Rem0o/FanControl.HWInfo works.

**Reply by Rem0o** [reply]
See https://github.com/Rem0o/FanControl.Releases/wiki/Plugins

**Comment by JajajTec** [comment]
I sadly only know the utmost basic about making my own (can open and change a name in vs thats it). But imma work myself into it and see if i manage to do it. Imma write here if i need more help is that oke?

**Reply by Rem0o** [reply]
Well I'm not here to teach you how to code.  I can assist with integration questions though. 
You could also use the "file" custom sensor to add a new sensor into FanControl.

Either way, If you want to interop an other software with FanControl, it will require you some dev knowledge, no way around it.

**Reply by JajajTec** [reply]
Wait i can point the file sensor to a registry? Because when i tryed to set the file sensor to a path it always resets to the fancontroll directory…

**Reply by Rem0o** [reply]
> Wait i can point the file sensor to a registry

Not exactly. The file can act as a bridge. You could have some program reading the registry and write to the file, this way FanControl wil natively pick it up.

**Reply by JajajTec** [reply]
Hmm good idea. That would be much simpler. Imma try that!
